### ISSUE TITLE
[FEATURE] Investigate blocking manual creation of ref_ids

### ISSUE THREAD

**Issue by lorawoodford** [Smithsonian/caas_aspace_refid#15]
**Is your feature request related to a problem? Please describe.**
As identified during the test party, there are several routes by which users can manually create or update ref_ids in such a way that these ref_ids are not tracked in the new `caas_aspace_refid` table in the db (leading to potential future collisions of ref_ids when aos are created via the UI).  These paths are:
- Creating an AO with ref_id via the API;
- Updating an AO with a ref_id via the API;
- Creating a new AO via the "Load via Spresheet" importer;
- Updating existing AOs via a "Spreadsheet Bulk Update Job"; or
- Creating new AOs via EAD import.

**Describe the solution you'd like**
I'd like to investigate what would be involved in blocking all these different workflows from accepting the ref_ids they are given, and instead default to the built-in plugin behavior for autogenerating ref_ids.

**Describe alternatives you've considered**
We may just accept the existing behavior given how sparsely these pathways to creating/updating AOs are used, but we're investigating options.

**Additional context**
This is intended as a quick investigation.  If a path forward seems clear and involves a low level of effort, we'll proceed; If not, we will document these potential pitfalls, but move on.

**Comment by lorawoodford** [comment]
After reviewing the code and how involved this would be to address via the plugin (very), my recommendation is to leave the plugin as-is, but I also recommend the following (to be discussed as a team, probably on 11/25):

- **API** It is assumed that - at this time - only very advanced users (likely, just CAAS team members) will ever be creating or updating archival objects via the API.  These users should be made aware of this issue and whatever documentation we have/will have related to API actions should specifically call out this warning.
- **Load via Spreadsheet Importer** At this time, it is highly unlikely our users will accidentally include ref_ids when using the Load Via Spreadsheet functionality as we currently point users to custom SI import templates.  Both the minimal and enhanced templates do not provide a ref_id column.  While we may move away from pointing users to Confluence for templates in the future, we can still ensure that the templates users retrieve from the application do not include the ref_id column.
- **Spreadsheet Bulk Update Job** I recommend we take the minimal approach of (temporarily) forking the bulk upload plugin we use with v3.3.1 to either deactivate/gray out the "ref_id" field on the download spreadsheet page, or outright remove it as a selectable field (there are plusses and minuses to both approaches).  We know we will be removing this plugin after our migration to ArchivesSpace v4.0, and at that time we can create a light lift plugin that will similarly gray out or remove the ref_id field (I have already tested this in 4.0 locally).  Essentially, this gets us to the same place we are with the Load Spreadsheet job wherein it will become incredibly unlikely that a user would ever attempt to run an import job that contains ref_ids and, therefore, modify existing ref_ids in a way that will create issues in the future.  (If the ability to export ref_ids in bulk is a feature we want to maintain, we could alternatively restrict that checkbox to select permissions/groups)
- **EAD Import** For this, I suggest we just keep things as they are and address issues as they arise/if they arise.  It seems like importing EAD is not a heavily used means of getting new resources into ArchivesSpace, and, even if it were, the odds of <c> id's being constructed to perfectly match the pattern used by the autogeneration plugin (leading to future collisions) seems very low.  This can likely be tackled via user education and documentation.

So, TLDR proposed future work:
- Create/update API documentation
- After discussing, fork and modify the bulk import plugin
- Update EAD import documentation

**Comment by fordmadox** [comment]
Agreed on all parts.  I'd also add that our department could / would take the lead on any EAD imports at this stage (which is what we did when adding CHSDM records), since I don't think that any units would be doing that otherwise.  Assuming that's correct, then the EAD imports and API updates can be handled the same way moving forward.

**Comment by lorawoodford** [comment]
Closing as complete.  Remaining work in https://github.com/Smithsonian/as_spreadsheet_bulk_updater/issues/1
